app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,60 @@
+class BookmarksController < ApplicationController
+  def show
+    @bookmark = Bookmark.find(params[:id])
+  end
+
+  def new
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = Bookmark.new
+  end
+
+  def edit
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+  end
+
+  def update
+    topic = Topic.find(params[:topic_id])
+    @bookmark = topic.bookmarks.find(params[:id])
+    if @bookmark.update(bookmark_params)
+      flash[:notice] = 'bookmark was saved successfully.'
+      redirect_to topic
+    else
+      flash.now[:alert] = @bookmark.errors.full_messages.first
+      render :edit
+    end
+  end
+
+  def create
+    @topic = Topic.find(params[:topic_id])
+    @bookmark = @topic.bookmarks.build(bookmark_params)
+    # @bookmark.user = current_user
+    # @new_bookmark = Bookmark.new
+    if @bookmark.save
+      flash[:notice] = 'bookmark was saved successfully.'
+      redirect_to @topic
+    else
+      flash.now[:alert] = 'Error creating bookmark. Please try again.'
+      render :new
+    end
+  end
+    def destroy
```

Comment:
Indentation seems off here. Usually want to add a new line between each method as well.